app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt

```diff
@@ -164,6 +163,16 @@ abstract class BaseBrowserFragment : Fragment(), UserInteractionHandler, Session
 
         initializeEngineView(toolbarHeight)
 
+        browserAnimator = BrowserAnimator(
+            fragment = WeakReference(this),
+            viewLifeCycleScope = viewLifecycleOwner.lifecycleScope,
+            engineView = engineView,
+            swipeRefresh = swipeRefresh,
```

Comment:
If we're taking the `WeakReference` route here, we'll need to wrap these two views as well.